src/Hosting.Services/HostBuilderExtensions.cs

```diff
@@ -97,29 +99,24 @@ private static IHost BuildServiceFabricService<TRunner, TService, TContext>(
 							.AddOmexServiceFabricDependencies<TContext>()
 							.AddSingleton<IOmexServiceRegistrator, TRunner>()
 							.AddHostedService<OmexHostedService>();
+						collection.TryAddTransient<IServiceContext, EmptyServiceContext>();
+						collection.TryAddTransient<IExecutionContext, BaseExecutionContext>();
```

Comment:
suggest adding a comment before these to indicate why they are added via TryAddTransient
```suggestion
						// If specific context constructs are not registered, add defaults for telemetry purposes
						collection.TryAddTransient<IServiceContext, EmptyServiceContext>();
						collection.TryAddTransient<IExecutionContext, BaseExecutionContext>();
```